### PR TITLE
feat: implement NIST P-256 and P-384 curves

### PR DESCRIPTION
**Pull request by ivokub**
These curves are fairly popular in practical applications (certificates, TLS, ID cards, signatures) and imo its better to have included so that users would implement correctly.

As these curves are supported by Go standard library, then do not need to add anything in gnark-crypto and can perform precomputation+tests using standard library only. So no more imports.

Is based on top of https://github.com/ConsenSys/gnark/pull/696, so it is better to have that reviewed/merged first.

Thanks to @SherLzp for initial contribution.

### CODE REVIEW

**Comment by ivokub** [comment]
Oh, I didn't realise this is already done in https://github.com/ConsenSys/gnark/pull/610. Will see if I can attribute the contributor by merging the implementations.

File: std/math/emulated/params.go [hunk 1]
```diff
@@ -3,108 +3,32 @@ package emulated
...
-// 64bits for modulus
-// 0x73eda753299d7d483339d80809a1d80553bda402fffe5bfeffffffff00000001.
-// This is the scalar field of the BLS12-381 curve.
-type BLS12381Fr struct{}
-
-func (fp BLS12381Fr) NbLimbs() uint     { return 4 }
-func (fp BLS12381Fr) BitsPerLimb() uint { return 64 }
-func (fp BLS12381Fr) IsPrime() bool     { return true }
-func (fp BLS12381Fr) Modulus() *big.Int { return ecc.BLS12_381.ScalarField() }
+type Goldilocks = emparams.Goldilocks
```
Comment (yelhousni) [inline std/math/emulated/params.go:27]: hmm so either enforce P256 and P384 types here or remove others but I don't have a strong opinion

**Review by yelhousni — APPROVED** [review]
Definitely a great contribution to gnark std for users! LGTM 👍

File: std/math/emulated/params.go (hunk 1 above)
Comment (ivokub) [reply std/math/emulated/params.go:27]: Included alias here.

**Pull request merged**